### PR TITLE
CT5UP-306: Check npm version and show warning for v.5

### PR DESCRIPTION
**Pull request by NataliaMarkaki**

### CODE REVIEW

File: cli/commands/create-app.js [hunk 1]
```diff
@@ -65,6 +66,21 @@ module.exports = {
...
+
+      const npmVersion = fs
+        .readdirSync(
+          path.join(globalModules, "..", "..", "npm-cache", "npmlog")
+        )
+        .pop();
+
+      if (npmVersion.startsWith("5")) {
+        console.log(
+          `You may have problems using some of the tooling with your current version of npm. Whilst npm address these issues we recommend using npm 4`
```
Comment (stevesouth) [inline cli/commands/create-app.js]: You may have problems using local libraries with your current version of npm. Until npm have addressed this issue we recommend using npm 4.

Comment (NataliaMarkaki) [reply cli/commands/create-app.js]: thank you @stevesouth ! Will change it now.

**Comment by briandipalma** [comment]
I'd like us to experiment with an alternative approach before continuing with this one. Could we not add a `preinstall` script to the webpack-config package that runs an npm version check script? `process.env` might contain the npm version. Documentation on scripts is available here https://docs.npmjs.com/misc/scripts @NataliaMarkaki

File: cli/package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     "auto-load": "^2.1.0",
     "chalk": "^1.1.3",
     "copy-template-dir": "^1.3.0",
-    "fmt": "^0.5.0",
+    "fmt": "^0.5.0",
```
Comment (briandipalma) [inline cli/package.json]: Delete this extra whitespace.

File: webpack-config-app/package.json [hunk 3]
```diff
@@ -5,6 +5,9 @@
   "main": "index.js",
   "author": "",
   "license": "ISC",
+  "scripts": {
+    "postinstall": "node post-install.js"
```
Comment (briandipalma) [inline webpack-config-app/package.json]: post install OK? Does npm 5 not error out before post install? Did preinstall not work?

File: webpack-config-app/post-install.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+const chalk = require("chalk");
+const exec = require("child_process").exec;
+
+exec("npm -v", (execError, stdin, stderr) => {
```
Comment (briandipalma) [inline webpack-config-app/post-install.js]: No need to change this but it's easier to use the Sync methods.

File: webpack-config-app/package.json (hunk 3 above)
Comment (NataliaMarkaki) [reply webpack-config-app/package.json]: I had the exact opposite happening. Pre install was throwing errors while post install was alright. Tested in both Mac and Windows.

**Pull request merged**